src/sock.c

```diff
@@ -374,3 +365,14 @@ int getpeer_information (int fd, char *ipaddr, char *string_addr)
         return getnameinfo ((struct sockaddr *) &sa, salen,
                             string_addr, HOSTNAME_LENGTH, NULL, 0, 0);
 }
+
+/*
+ * Return a human readable error for getaddrinfo() and getnameinfo().
+ */
+static const char * get_gai_error (int n)
```

Comment:
cool. i guess though you need this function to the top so the callers can see the declaration

Reply:
Oh man, my C has rusted.

Reply:
Is it possible to squash all those commits directly on Github? If not I'd better create another PR.

Reply:
yeah, you can squash them locally, then check everything's right with e.g. gitk (and a testbuild), then force-push